drivers/hmis/app/models/hmis/hud/organization.rb

```diff
@@ -36,7 +36,12 @@ class Hmis::Hud::Organization < Hmis::Hud::Base
 
     search_term.strip!
     query = "%#{search_term.split(/\W+/).join('%')}%"
-    where(o_t[:OrganizationName].matches(query).or(o_t[:id].eq(search_term)).or(o_t[:organization_id].eq(search_term)))
+
+    where(
+      o_t[:OrganizationName].matches(query).
+        or(o_t[:id].eq(possibly_pk?(search_term) ? search_term : '')).
+        or(o_t[:organization_id].eq(search_term)),
+    )
```

Comment:
This feels wrong to be checking `o_t[:id].eq('')` as a no-op. I think it would be clearer to just omit that condition if `search_term` is not a possible pk. If you build up an array of conditions, you can chain them like this `where([ cond1, cond2, cond3].inject(&:or))`